packages/api/src/lib/mmt.ts

```diff
@@ -419,46 +419,47 @@ export type LoRelationToDimAndConceptPair = (typeof ALL_DIM_CONCEPT_PAIR)[number
 export type LoRelationToNonDimConcept = (typeof ALL_NON_DIM_CONCEPT)[number];
 export type AllLoRelationTypes = (typeof ALL_LO_RELATION_TYPES)[number];
 
-export const getSparqlQueryForLoRelationToDimAndConceptPair = (
-  uri: string,
-  relationType: LoRelationToDimAndConceptPair
-) => {
+export const getSparqlQueryForLoRelationToDimAndConceptPair = (uri: string) => {
   if (!uri) {
     console.error('URI is absent');
     return;
   }
-  const query = `
-        PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
-        PREFIX ulo: <http://mathhub.info/ulo#>
-
-        SELECT ?learningObject ?obj1 (GROUP_CONCAT(CONCAT(STR(?relType), "=", STR(?obj2)); SEPARATOR="; ") AS ?relatedData)
-        WHERE {
-            ?learningObject ulo:${relationType} ?obj1 .
-            ?obj1 ?relType ?obj2 .
-            FILTER(CONTAINS(STR(?learningObject), "${uri}")).
-        }
-        GROUP BY ?learningObject ?obj1
-      `;
+  const query = `PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+                PREFIX ulo: <http://mathhub.info/ulo#>
+
+                SELECT ?learningObject ?relation ?obj1 (GROUP_CONCAT(CONCAT(STR(?relType), "=", STR(?obj2)); SEPARATOR="; ") AS ?relatedData)
```

Comment:
I think `?obj1 ` can be skipped

```suggestion
                SELECT ?learningObject ?relation (GROUP_CONCAT(CONCAT(STR(?relType), "=", STR(?obj2)); SEPARATOR="; ") AS ?relatedData)
```

Reply:
Yes obj1 is not required , even learninngObject  variable is not required , wrote just to get clear view of which data is connected to which one and by what means . only using relatedData and relation we can do our work. and similarly in second type of uris as well  we can skip retrieving learningObject . so should i remove these two or proceed with as it is? or deal with it in next pr..